openpype/modules/slack/plugins/publish/integrate_slack_api.py

```diff
@@ -0,0 +1,146 @@
+import os
+try:
+    from slackclient import SlackClient
+    python2 = True
+except ImportError:
+    python2 = False
+    from slack_sdk import WebClient
+    from slack_sdk.errors import SlackApiError
+
+import pyblish.api
+from openpype.lib.plugin_tools import prepare_template_data
+
+
+class IntegrateSlackAPI(pyblish.api.InstancePlugin):
+    """ Send message notification to a channel.
+
+        Triggers on instances with "slack" family, filled by
+        'collect_slack_family'.
+
+        Expects configured profile in
+        Project settings > Slack > Publish plugins > Notification to Slack.
+
+        If instance contains 'thumbnail' it uploads it. Bot must be present
+        in the target channel.
+
+        Message template can contain {} placeholders from anatomyData.
+    """
+    order = pyblish.api.IntegratorOrder + 0.499
+    label = "Integrate Slack Api"
+    families = ["slack"]
+
+    optional = True
+
+    def process(self, instance):
+        message_templ = instance.data["slack_message"]
+
+        fill_pairs = (
```

Comment:
I don't it's good idea to use 3 varianted keys/values here (normal, capitalized and upper) as it's a lot limiting and adds different formatting keys than are in whole settings system (e.g. using `project_name` instead of `project[name]`). I would just create copy of `anatomyData` and add "few specified keys" that can have these capitalized variants (And actually can't image which keys make sence to have in these variants?).

Reply:
This was reduced to predefined keys...